### ISSUE TITLE
High 'conf loss'

### ISSUE THREAD

**Issue by soumendukrg** [marvis/pytorch-yolo2#43]
I am getting a very **high confidence loss** from **region_loss.py** when training on both _voc_dataset_ as well as custom on first few epochs. I don't have a GPU, so I have to train on CPU, I am only able to see results at epoch 8 till now. I did have to change the calculation of **nProposals** as I was getting some error earlier. Also, I am using python 3 instead of python 2 and I had to make some typecasting to int in some files. 

The following is the output of training. My question is it normal that **conf loss** will be this high value initially and after some epochs it will reduce or is there any other issue?

Moreover, the no of proposals has remained constant at 8450, I can't understand the reason.

`8094: nGT 13, recall 11, proposals 8450, loss: x 0.735876, y 0.195984, w 1.357148, h 0.712104, conf 1049.567871, cls 15.311121, total 1067.880127
8104: nGT 12, recall 9, proposals 8450, loss: x 0.556932, y 0.568213, w 0.896144, h 0.685480, conf 1051.043213, cls 7.622620, total 1061.372681
8114: nGT 10, recall 10, proposals 8450, loss: x 0.346231, y 0.704523, w 0.134786, h 0.436545, conf 1047.272461, cls 7.384933, total 1056.279419
8124: nGT 14, recall 9, proposals 8450, loss: x 0.521500, y 0.579621, w 2.376108, h 1.173029, conf 1048.203857, cls 11.298470, total 1064.152588
8134: nGT 18, recall 9, proposals 8450, loss: x 0.844499, y 0.464092, w 4.488868, h 1.119425, conf 1050.343750, cls 11.339650, total 1068.600220
8144: nGT 13, recall 8, proposals 8450, loss: x 0.490713, y 0.654974, w 1.219727, h 1.798023, conf 1051.956543, cls 10.618993, total 1066.739014
8154: nGT 16, recall 16, proposals 8450, loss: x 0.525403, y 0.471665, w 0.442946, h 0.270117, conf 1047.184326, cls 15.675910, total 1064.570312
8164: nGT 13, recall 9, proposals 8450, loss: x 0.470466, y 0.465599, w 0.894855, h 0.248689, conf 1048.053223, cls 13.963310, total 1064.096069
8174: nGT 10, recall 10, proposals 8450, loss: x 0.468166, y 0.291908, w 0.192603, h 0.296834, conf 1048.890259, cls 5.321609, total 1055.461426`

Please suggest some idea on how to proceed.

**Comment by soumendukrg** [comment]
The problem is solved now. I changed some code for porting from python2 to python3 and that was giving the high conf loss and that many high no of proposals.
